src/main/java/org/sonarlint/intellij/SonarLintIntelliJClient.kt

```diff
@@ -222,6 +220,28 @@ object SonarLintIntelliJClient : SonarLintClient {
         showBalloon(null, params.text, convert(params.type))
     }
 
+    override fun log(params: LogParams) {
+        if (params.configScopeId != null) {
+            var project = BackendService.findModule(params.configScopeId!!)?.project ?: BackendService.findProject(params.configScopeId!!)
```

Comment:
There should be a better way to do this in Kotlin, since you have already checked for null in the if, I don't expect to see the use of `!!`

Reply:
There is also a (small) chance that no projects can be found at all (for example if the backend try to log, but the project has already been closed on IDE side). In this case you should fallback to the global log output.